module-bluetooth/Bluetooth/interface/profiles/SCO/ScoUtils.cpp

```diff
@@ -15,19 +15,23 @@ namespace bluetooth::sco::utils
     void sendZeros(hci_con_handle_t scoHandle)
     {
         if (scoHandle == HCI_CON_HANDLE_INVALID) {
+            LOG_ERROR("Invalid scoHandle!");
             return;
         }
-        constexpr auto offset         = 3;
+        constexpr auto packetDataOffset = 3;
         constexpr auto lengthPosition = 2;
         const auto scoPacketLength    = hci_get_sco_packet_length();
-        const auto scoPayloadLength   = scoPacketLength - offset;
+        const auto scoPayloadLength     = scoPacketLength - packetDataOffset;
 
         hci_reserve_packet_buffer();
-        auto *scoPacket = hci_get_outgoing_packet_buffer();
-        std::fill(scoPacket + offset, scoPacket + sizeof(scoPacket), 0);
+        auto scoPacket     = hci_get_outgoing_packet_buffer();
+        auto scoPayloadPtr = scoPacket + packetDataOffset;
+
+        memset(scoPayloadPtr, 0, scoPayloadLength); // zero-fill the payload
```

Comment:
:+1:

Reply:
This was changed to fix endlessly looping sound in BT speaker after switching to internal Pure's speaker during call ;)